File: common/src/main/java/com/ssblur/scriptor/entity/renderers/ColorfulSheepRenderer.java
package com.ssblur.scriptor.entity.renderers;

import com.ssblur.scriptor.entity.ColorfulSheep;
import com.ssblur.scriptor.entity.renderers.layer.ColorfulSheepFurLayer;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.model.SheepModel;
import net.minecraft.client.model.geom.ModelLayers;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.client.renderer.entity.MobRenderer;
import net.minecraft.resources.ResourceLocation;

@Environment(EnvType.CLIENT)
public class ColorfulSheepRenderer extends MobRenderer<ColorfulSheep, SheepModel<ColorfulSheep>> {
  private static final ResourceLocation SHEEP_LOCATION = new ResourceLocation("textures/entity/sheep/sheep.png");

  public ColorfulSheepRenderer(EntityRendererProvider.Context context) {
    super(context, new SheepModel<>(context.bakeLayer(ModelLayers.SHEEP)), 0.7F);
    this.addLayer(new ColorfulSheepFurLayer(this, context.getModelSet()));
  }

  public ResourceLocation getTextureLocation(ColorfulSheep sheep) {
    return SHEEP_LOCATION;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/entity/renderers/ColorfulSheepRenderer.kt
package com.ssblur.scriptor.entity.renderers

import com.ssblur.scriptor.entity.ColorfulSheep
import com.ssblur.scriptor.entity.renderers.layer.ColorfulSheepFurLayer
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.model.SheepModel
import net.minecraft.client.model.geom.ModelLayers
import net.minecraft.client.renderer.entity.EntityRendererProvider
import net.minecraft.client.renderer.entity.MobRenderer
import net.minecraft.resources.ResourceLocation

@Environment(EnvType.CLIENT)
class ColorfulSheepRenderer(context: EntityRendererProvider.Context):
  MobRenderer<ColorfulSheep, SheepModel<ColorfulSheep>?>(
    context, SheepModel(
      context.bakeLayer(
        ModelLayers.SHEEP
      )
    ), 0.7f
  ) {
  init {
    this.addLayer(ColorfulSheepFurLayer(this, context.modelSet))
  }

  override fun getTextureLocation(sheep: ColorfulSheep): ResourceLocation {
    return SHEEP_LOCATION
  }

  companion object {
    private val SHEEP_LOCATION: ResourceLocation = ResourceLocation.parse("textures/entity/sheep/sheep.png")
  }
}
